src/textual/app.py

```diff
@@ -1630,8 +1630,22 @@ def _fatal_error(self) -> None:
 
     def _print_error_renderables(self) -> None:
         """Print and clear exit renderables."""
-        for renderable in self._exit_renderables:
-            self.error_console.print(renderable)
+        error_count = len(self._exit_renderables)
+        if "debug" in self.features:
+            for renderable in self._exit_renderables:
+                self.error_console.print(renderable)
+            if error_count > 1:
+                self.error_console.print(
+                    f"\n[b]NOTE:[/b] {error_count} errors show above.", markup=True
+                )
+        elif self._exit_renderables:
+            self.error_console.print(self._exit_renderables[0])
+            if error_count > 1:
+                self.error_console.print(
+                    f"\n[b]NOTE:[/b] 1 of {error_count} errors show. Run with [b]--dev[/] to see all errors.",
+                    markup=True,
+                )
```

Comment:
This makes sense. At the same time, the error(s) may show up unexpectedly and the user may not be able to reproduce them. Would it make even more sense to write the remaining errors to a log?

Reply:
If logging is enabled, these will still go to the logs.